sklearn/mixture/dpgaussianmixture.py

```diff
@@ -0,0 +1,245 @@
+"""Dirichlet Process Bayesian Gaussian Mixture Model."""
+import numpy as np
+from scipy.special import digamma, gammaln
+
+from ..utils.validation import check_is_fitted
+from .bayesianmixture import BayesianGaussianMixture, check_shape
+
+
+def _log_beta_norm(a, b):
+    """The log of the normalization of Beta distribution."""
+    return gammaln(a+b) - gammaln(a) - gammaln(b)
+
+
+class DirichletProcessGaussianMixture(BayesianGaussianMixture):
+
+    """Dirichlet Process Bayesian Gaussian Mixture Model.
+
+    An infinite mixture model with the Dirichlet Process as a prior
+    distribution on the number of clusters. In practice the
+    approximate inference algorithm uses a truncated distribution with
+    a fixed maximum number of components, but almost always the number
+    of components actually used depends on the data.
+
+    Stick-breaking Representation of a Gaussian mixture model
+    probability distribution. This class allows for easy and efficient
+    inference of an approximate posterior distribution over the
+    parameters of a Gaussian mixture model with a variable number of
+    components (smaller than the truncation parameter n_components).
+
+    Read more in the :ref:`User Guide <dpgmm>`.
+
+    Parameters
+    ----------
+    n_components: int, default 1.
+        Number of mixture components.
+
+    precision_type: string, default to 'full'.
+        String describing the type of covariance parameters to
+        use.  Must be one of 'spherical', 'tied', 'diag', 'full'.
+
+    tol : float, default 1e-6
+        Convergence threshold.
+
+    reg_covar : float, defaults to 0.
+        Non-negative regularization to the diagonal of covariance.
+
+    n_iter : int, default to 100.
+        Maximum number of iterations to perform before convergence.
+
+    n_init : int, default to 1.
+        Number of initializations to perform. The best results is kept.
+
+    params : string, default to None.
+        Controls which parameters are updated in the initialization
+        process.  Can contain any combination of 'w' for weights,
+        'm' for means, and 'c' for covars.  Defaults to 'wmc'.
+
+    init_params : string, defaults to 'kmeans'.
+        Controls how parameters are initialized unless the prior
+        parameters are provided by users. It should be one of 'kmeans',
+        'random', None. If it is not None, the responsibilities and model
+        parameters are initialized by the chosen method.
+
+    gamma_prior: float, default to 0.1.
```

Comment:
Sorry if I've misunderstood your notation, but I think the expected number of mixture components grows as `alpha_prior * log(N)`.